### PR TITLE
Add Infinity end-to-end text-to-image pipeline

### LINKED ISSUE #5546: Add Infinity end-to-end text-to-image pipeline

### PR DESCRIPTION
**Pull request by meenakshiramanathan1**
### Ticket

- Fixes [#5546](https://github.com/tenstorrent/tt-xla/issues/5546)

### Problem description

To add Infinity end-to-end text-to-image pipeline.

### What's changed

- Adds a self-contained, end-to-end Infinity 2B text-to-image pipeline at infinity/pytorch/src/pipeline.py, so consumers (the tt-xla nightly e2e test and the perf benchmark) can import one shared pipeline instead of each re-implementing the generation loop.

- Exposes InfinityConfig and InfinityPipeline (setup → load_models / shard_to_tt / _build_scale_schedule, and generate(prompt, seed) -> image), reimplementing the model's autoregressive_infer_cfg as an explicit next-scale-prediction loop (transformer forward + multinomial sampling + BSQ-VAE code accumulation per scale, then a single VAE decode).

- Establishes the CPU/TT device split: the 2B transformer runs on Tenstorrent, 8-way tensor-parallel sharded (mesh (1, 8), Megatron head-parallel attention from loader.load_shard_spec); the T5-XL text encoder, multinomial sampling, and BSQ-VAE decode stay on CPU.

- Implements sequential classifier-free guidance — cond and uncond are run as two batch-1 forwards per scale and combined on the logits. (A batch-2 stacked forward makes the attention score matmul all-gather the heads / de-shard and OOM at the final 1M scale; batch-1 keeps the score head-sharded.)

- Implements fp32 LayerNorm via an explicit mean/var/rsqrt decomposition (_force_fp32_layernorm). The bf16 fused ttnn.layer_norm loses precision on the mid/late layers' outlier activations, which autoregressive sampling amplifies into a noise image; the fp32 decomposition restores per-scale PCC to ~1.0 and yields a coherent image. (A plain F.layer_norm(x.float()) does not help — it folds back to bf16.)

- Result: generates the model's native 1024×1024 (1M) output with full classifier-free guidance, sharded end-to-end on Tenstorrent.

Verified per-component PCC against an fp32 CPU reference on the real pipeline tensors.

Details from log:
Nightly e2e test `tests/torch/models/infinity/test_infinity_pipeline.py::test_infinity_pipeline` — **PASSED** (41m30s).

| Stage | Device | Time |
|-------|--------|-----------|
| T5-XL text encode | CPU | ~44s |
| Next-scale prediction loop (13 scales, CFG) | TT (bf16, 8-way TP) | ~39m55s |
| BSQ-VAE decode | CPU | ~9s |
| **Total** | | **2490.91s** |

```
[STAGE] T5 text encode (CPU): start        03:01:44
[STAGE] T5 text encode (CPU): done         03:02:28
[STEP] scale  1/13 (1,  1,  1)             03:02:28
[STEP] scale  2/13 (1,  2,  2)             03:07:37
[STEP] scale  3/13 (1,  4,  4)             03:09:46
[STEP] scale  4/13 (1,  6,  6)             03:11:42
[STEP] scale  5/13 (1,  8,  8)             03:14:28
[STEP] scale  6/13 (1, 12, 12)             03:16:59
[STEP] scale  7/13 (1, 16, 16)             03:19:53
[STEP] scale  8/13 (1, 20, 20)             03:22:41
[STEP] scale  9/13 (1, 24, 24)             03:25:36
[STEP] scale 10/13 (1, 32, 32)             03:28:42
[STEP] scale 11/13 (1, 40, 40)             03:31:56
[STEP] scale 12/13 (1, 48, 48)             03:35:12
[STEP] scale 13/13 (1, 64, 64)             03:38:45
[STAGE] BSQ-VAE decode (CPU): start        03:42:23
[STAGE] BSQ-VAE decode (CPU): done         03:42:32
```

Final image generated from pipeline:
PROMPT = "A fantasy landscape with mountains and rivers"
infinity_2b_output

### CODE REVIEW

File: infinity/pytorch/src/pipeline.py [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+DTYPE = torch.bfloat16
+
+
+def _enable_spmd() -> None:
+    """Enable torch_xla SPMD (shardy) — required before any device op.
+
+    Mirrors ``tests/infra/utilities/torch_multichip_utils.enable_spmd`` but is
+    inlined so this module carries no tt-xla test dependency.
+    """
+    os.environ["CONVERT_SHLO_TO_SHARDY"] = "1"
```
Comment (nvukobratTT) [inline infinity/pytorch/src/pipeline.py:63]: Do we need this?

**Review by nvukobratTT — APPROVED** [review]

File: infinity/pytorch/src/pipeline.py (hunk 1 above)
Comment (meenakshiramanathan1) [reply infinity/pytorch/src/pipeline.py:63]: This pipeline shards the transformer 8-way. CONVERT_SHLO_TO_SHARDY=1 is exactly what enables the StableHLO→Shardy conversion pass in the pytorch-xla fork. So it's required because we shard here.  It must run before xr.use_spmd() and any device op , and the pipeline is what performs the sharding/device ops. It's inlined rather than imported from torch_multichip_utils.enable_spmd so the tt_forge_models submodule carries no tt-xla test dependency.
